### ISSUE TITLE
Character description is different than in original

### ISSUE THREAD

**Issue by fenuks** [gemrb/gemrb#682]
#### Bug description
In new character window, in appearance tab, character description misses some information, in one case skill value has different value than in original, there are present section titles with no information. Below list of inconsistencies with regard to original for pre-made characters. To avoid flooding this issue with screenshots with side-by-side comparison, I added them to [this imgur collection](https://imgur.com/a/WI2YXGV). It's easier to grasp what I am talking about by examining screenshots.

### Bard
- pocket theft skill in IE shows 90%, and 65% in GemRB
- in IE proficiencies are shown after known mage spells
- known mage spells miss plenty of spells

### Druid
- empty class enemy section
- empty skills section
- missing priest spells

### Paladin
- strength in original is formatted as `18/39`

GemRB version: latest git

#### Steps to reproduce
1. Go to 'New Game -> Import'
2. Import one of characters
3. See differences with original

#### Expected behavior
Description should be more close to these in original (especially in case of spells).

**Comment by lynxlynxlynx** [comment]
The skill % is probably missing the dex bonus we apply in the engine. Might be too early here.
The missing spell thing requires an investigation while the rest is just about reordering and enabling.

**Comment by fenuks** [comment]
I plan to correct these problems. Question, how do you debug python GUI scripts? I know there is built-in python console in game, but I would like to use proper debugger like GDB or pdb.

**Comment by MarcelHB** [comment]
I think I tried that long time ago: https://pypi.org/project/rpdb/

**Comment by lynxlynxlynx** [comment]
For these sorts of issues, I suggest you don't bother (import pdb; pdb.set_trace() # otherwise insert when needed) and just add debug prints as needed. PDB is quite frustrating in my experience and these issues are mostly not algorithmic in a sense that you'd need to inspect  a lot of state.

btw, for the paladin thing you just need to append IE_STREXTRA.

**Comment by fenuks** [comment]
I like to have debugger prompt for interactive experimenting as if it was interpreter, as I am not familiar with GUIScripts. That would just ease learning curve, especially that changing python script requires game re-run, which makes iteration process fairly slow.

> btw, for the paladin thing you just need to append IE_STREXTRA.

Thanks.

**Comment by lynxlynxlynx** [comment]
On other thing I forgot to mention: most of these bugs are not present in bg2, likely also bg1, so you can inspect their chargen files for hints and code.

**Comment by bradallred** [comment]
if you really want to debug then get a newish version of gdb and follow [this](https://devguide.python.org/gdb/)

**Comment by fenuks** [comment]
I figured out why extra strength is not shown for Icewind Dale, even though python script has support for it. It turns out that unhardcored `classes.2da` are not present for `how`, only for `iwd`, which file contains `STREXTRA` value GUI relies upon. 

Without `classes.2da` `CommonTables.Classes.GetValue (ClassName, "STREXTRA", GTV_INT)` evaluates to None, and `CommonTables.Classes.GetValue (ClassName, "STREXTRA", GTV_STR)` to `*`.

Beside that, something like `HasStrExtra = CommonTables.Classes.GetValue (ClassName, "STREXTRA", GTV_INT)` is missing after importing character. That is easy to fix, but I can see that in case of `bg`, `HasStrExtra` flag is not used, instead of that, extra strength value is expected to be larger than 0, [like so](https://github.com/gemrb/gemrb/blob/997ba6d289c8b07c86ee9186079df5d8f492f987/gemrb/GUIScripts/bg2/CharGenCommon.py#L166).

Which approach should be used?

**Comment by bradallred** [comment]
I wish we used the same nested structure for iwd/how as we do at the root level. then how could just inherit from iwd.

**Comment by lynxlynxlynx** [comment]
Of course how has classes.2da, that's both trivial to check and crucial. It appears I did forget it in this commit though: 0a0cb621. Fixed.

Extra strength has to be over 0, negatives don't make sense and 0 was never displayed. "00" is 100. When it comes to displaying it in the overview, it's not about whether the class supports it or not any more, just the actual value generated or imported, so to fix the latter case it's indeed better to just check the stat. Importing in general is not something we've been very careful about, so there's bound to be more cosmetic problems like this, also in other games.

@bradallred: they do use the same scripts, though that was not always the case.
